src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -91,195 +62,100 @@ public string OutputFilePath
 
         /// <summary>
         /// The options used when parsing documents for this project.
         /// </summary>
-        public ParseOptions ParseOptions
-        {
-            get
-            {
-                return _projectState.ParseOptions;
-            }
-        }
+        public ParseOptions ParseOptions => _projectState.ParseOptions;
 
         /// <summary>
         /// Returns true if this is a submission project.
         /// </summary>
-        public bool IsSubmission
-        {
-            get
-            {
-                return _projectState.IsSubmission;
-            }
-        }
+        public bool IsSubmission => _projectState.IsSubmission;
 
         /// <summary>
         /// True if the project has any documents.
         /// </summary>
-        public bool HasDocuments
-        {
-            get
-            {
-                return _projectState.HasDocuments;
-            }
-        }
+        public bool HasDocuments => _projectState.HasDocuments;
 
         /// <summary>
         /// All the document IDs associated with this project.
         /// </summary>
-        public IReadOnlyList<DocumentId> DocumentIds
-        {
-            get
-            {
-                return _projectState.DocumentIds;
-            }
-        }
+        public IReadOnlyList<DocumentId> DocumentIds => _projectState.DocumentIds;
 
         /// <summary>
         /// All the additional document IDs associated with this project.
         /// </summary>
-        public IReadOnlyList<DocumentId> AdditionalDocumentIds
-        {
-            get
-            {
-                return _projectState.AdditionalDocumentIds;
-            }
-        }
+        public IReadOnlyList<DocumentId> AdditionalDocumentIds => _projectState.AdditionalDocumentIds;
 
         /// <summary>
         /// All the documents associated with this project.
         /// </summary>
-        public IEnumerable<Document> Documents
-        {
-            get
-            {
-                return _projectState.DocumentIds.Select(GetDocument);
-            }
-        }
+        public IEnumerable<Document> Documents => _projectState.DocumentIds.Select(GetDocument);
 
-        public IEnumerable<TextDocument> AdditionalDocuments
-        {
-            get
-            {
-                return _projectState.AdditionalDocumentIds.Select(GetAdditionalDocument);
-            }
-        }
+        /// <summary>
+        /// All the additional documents associated with this project.
+        /// </summary>
+        public IEnumerable<TextDocument> AdditionalDocuments => _projectState.AdditionalDocumentIds.Select(GetAdditionalDocument);
+
+        /// <summary>
+        /// True if project information is complete. <see cref="ProjectInfo.IsComplete"/> for more info.
+        /// 
+        /// this is for internal usage only. for public usage, use <see cref="IsCompleteAsync(CancellationToken)"/>
```

Comment:
...

Ok, but as before *why* should internal code use this.  Why would internal code not use the other form?  Youv'e explained elsewhere, but this is an example of why the current code is confusing and difficult to reason about.